### PR TITLE
Replicate all of the LLVM linkage types in the repo.

### PR DESCRIPTION
**Pull request by MaggieYingYi**
This pull request is a part of the fix for <https://github.com/SNSystems/llvm-project-prepo/issues/40>.

### CODE REVIEW

File: include/pstore/mcrepo/compilation.hpp [hunk 1]
```diff
@@ -53,11 +53,15 @@ namespace pstore {
     namespace repo {
 
 #define PSTORE_REPO_LINKAGE_TYPES                                                                  \
-    X (append)                                                                                     \
-    X (common)                                                                                     \
-    X (external)                                                                                   \
-    X (internal)                                                                                   \
-    X (linkonce)
+    X (append_linkage)                                                                             \
```
Comment (paulhuggett) [inline include/pstore/mcrepo/compilation.hpp]: Do we need to have the word “linkage” added to the end of all of these? They're defined in a enum class named linkage_type so isn’t this redundant?

File: include/pstore/mcrepo/compilation.hpp [hunk 2]
```diff
@@ -53,11 +53,15 @@ namespace pstore {
...
+    X (common_linkage)                                                                             \
+    X (external_linkage)                                                                           \
+    X (private_linkage)                                                                            \
+    X (internal_linkage)                                                                           \
+    X (linkonceany_linkage)                                                                        \
```
Comment (paulhuggett) [inline include/pstore/mcrepo/compilation.hpp]: `link_once_any` perhaps?

File: include/pstore/mcrepo/compilation.hpp [hunk 3]
```diff
@@ -53,11 +53,15 @@ namespace pstore {
...
+    X (linkonceodr_linkage)                                                                        \
```
Comment (paulhuggett) [inline include/pstore/mcrepo/compilation.hpp]: `link_once_odr` perhaps?

File: include/pstore/mcrepo/compilation.hpp [hunk 4]
```diff
@@ -53,11 +53,15 @@ namespace pstore {
...
+    X (weakany_linkage)                                                                            \
```
Comment (paulhuggett) [inline include/pstore/mcrepo/compilation.hpp]: I‘m sure you get the idea…

File: include/pstore/mcrepo/compilation.hpp (hunk 1 above)
Comment (MaggieYingYi) [reply include/pstore/mcrepo/compilation.hpp]: The "private" is a keyword.  I had to borrow the format of the LLVM linkage type and add the `_linkage` suffix.

File: include/pstore/mcrepo/compilation.hpp (hunk 2 above)
Comment (MaggieYingYi) [reply include/pstore/mcrepo/compilation.hpp]: Ok, I will change it.

File: include/pstore/mcrepo/compilation.hpp (hunk 3 above)
Comment (MaggieYingYi) [reply include/pstore/mcrepo/compilation.hpp]: Ok, thanks.

File: include/pstore/mcrepo/compilation.hpp (hunk 4 above)
Comment (MaggieYingYi) [reply include/pstore/mcrepo/compilation.hpp]: Yes, I will change: 
1) `weakany_linkage` to `weak_any_linkage`
2) `weakodr_linkage` to `weak_odr_linkage`

File: include/pstore/mcrepo/compilation.hpp (hunk 1 above)
Comment (paulhuggett) [reply include/pstore/mcrepo/compilation.hpp]: Maybe use a different word? The LLVM sources define `PrivateLinkage` as "Like Internal, but omit from symbol table"; perhaps `internal_no_symbol`?

Comment (MaggieYingYi) [reply include/pstore/mcrepo/compilation.hpp]: Good point. I have updated the patch following your advice.

**Pull request merged**